pexconn.go

```diff
@@ -88,6 +88,15 @@ func (s *pexConnState) Share(postfn messageWriter) bool {
 
 // Recv is called from the reader goroutine
 func (s *pexConnState) Recv(payload []byte) error {
+	if !s.torrent.wantPeers() {
```

Comment:
This is tricky: The peer high and low water levels are designed for this, and addPeers should take care of that. Per my other comment, are we avoiding addPeers for performance costs?

Reply:
Indeed, and `wantPeer()` is one the parts of the high/low-watermak mechanism. Without it, PEX keeps feeding peers regardless whether the low watermark is reached; `addPeers` only cuts off at high watermark.